code/modules/mob/living/carbon/update_icons.dm

```diff
@@ -101,53 +96,59 @@
 
 /mob/living/carbon/update_inv_wear_mask()
 	remove_overlay(FACEMASK_LAYER)
-
 	if(istype(wear_mask, /obj/item/clothing/mask))
-
 		if(!(head && (head.flags_inv & HIDEMASK)))
-
 			var/image/standing = wear_mask.build_worn_icon(state = wear_mask.icon_state, default_layer = FACEMASK_LAYER, default_icon_file = 'icons/mob/mask.dmi')
-			overlays_standing[FACEMASK_LAYER]	= standing
-		return wear_mask
+			overlays_standing[FACEMASK_LAYER] = standing
+		update_hud_wear_mask(wear_mask)
+	apply_overlay(FACEMASK_LAYER)
 
 /mob/living/carbon/update_inv_back()
 	remove_overlay(BACK_LAYER)
 	if(back)
-
 		var/image/standing = back.build_worn_icon(state = back.icon_state, default_layer = BACK_LAYER, default_icon_file = 'icons/mob/back.dmi')
 		overlays_standing[BACK_LAYER] = standing
-		return back
-
+		update_hud_back(back)
+	apply_overlay(BACK_LAYER)
 
 /mob/living/carbon/update_inv_head()
 	remove_overlay(HEAD_LAYER)
 	if(head)
-
 		var/image/standing = head.build_worn_icon(state = head.icon_state, default_layer = HEAD_LAYER, default_icon_file = 'icons/mob/head.dmi')
 		overlays_standing[HEAD_LAYER] = standing
-		return head
+		update_hud_head(head)
+	apply_overlay(HEAD_LAYER)
 
 /mob/living/carbon/update_inv_handcuffed()
-	remove_overlay(HANDCUFF_LAYER)
-	clear_alert("handcuffed")
-	if(handcuffed)
-		drop_r_hand()
-		drop_l_hand()
-		stop_pulling()	//TODO: should be handled elsewhere
-		throw_alert("handcuffed", /obj/screen/alert/restrained/handcuffed, new_master = src.handcuffed)
-		if(hud_used)	//hud handcuff icons
-			var/obj/screen/inventory/R = hud_used.r_hand_hud_object
-			var/obj/screen/inventory/L = hud_used.l_hand_hud_object
-			R.overlays += image("icon"='icons/mob/screen_gen.dmi', "icon_state"="markus")
-			L.overlays += image("icon"='icons/mob/screen_gen.dmi', "icon_state"="gabrielle")
-		return 1
-	else
-		if(hud_used)
-			var/obj/screen/inventory/R = hud_used.r_hand_hud_object
-			var/obj/screen/inventory/L = hud_used.l_hand_hud_object
-			R.overlays = null
-			L.overlays = null
-
+	return
+
+
+//mob HUD updates for items in our inventory
+
+//update whether handcuffs appears on our hud.
+/mob/living/carbon/proc/update_hud_handcuffed()
+	if(hud_used)
+		var/obj/screen/inventory/R = hud_used.r_hand_hud_object
+		var/obj/screen/inventory/L = hud_used.l_hand_hud_object
+		if(R && L)
+			if(handcuffed)	//hud handcuff icons
+				R.overlays += image("icon"='icons/mob/screen_gen.dmi', "icon_state"="markus")
+				L.overlays += image("icon"='icons/mob/screen_gen.dmi', "icon_state"="gabrielle")
+			else
+				R.overlays = null
+				L.overlays = null
+
+//update whether our head item appears on our hud.
+/mob/living/carbon/proc/update_hud_head(obj/item/I)
+	return
+
+//update whether our mask item appears on our hud.
+/mob/living/carbon/proc/update_hud_wear_mask(obj/item/I)
+	return
+
+//update whether our back item appears on our hud.
+/mob/living/carbon/proc/update_hud_back(obj/item/I)
+	return

```

Comment:
These procs are here to handle the hud updates to do when we change our inventory, so these updates are not done directly in update_inv_xxx() proc, they now call our new procs to do that.